### PR TITLE
Expand and modernize documentation

### PR DESCRIPTION
**Pull request by magnusbaeck**
Replace old-fashioned "type => foo" conditional with an "if" conditional block in the example and generally expand the documentation to decrease the need for reading the source code to understand what the plugin does.

### CODE REVIEW

**Comment by ghost** [comment]
Jenkins standing by to test this. If you aren't a maintainer, you can ignore this comment. Someone with commit access, please review this and clear it for Jenkins to run; then  say 'jenkins, test it'.

**Comment by magnusbaeck** [comment]
@logstash-plugins/logstash-core, please review.

File: lib/logstash/outputs/exec.rb [hunk 1]
```diff
@@ -2,25 +2,36 @@
...
+# WARNING: Exercise great caution with `%{name}` field placeholders.
+# The contents of the field will be included verbatim without any
+# sanitization, i.e. any shell metacharacters from the field values
+# will be passed straight to the shell.
 class LogStash::Outputs::Exec < LogStash::Outputs::Base
 
   config_name "exec"
 
-  # Command line to execute via subprocess. Use dtach or screen to make it non blocking
+  # Command line to execute via subprocess. Use dtach or screen to
```
Comment (untergeek) [inline lib/logstash/outputs/exec.rb]: Maybe make this `dtach` and `screen` to indicate they are command-line processes?

File: lib/logstash/outputs/exec.rb [hunk 2]
```diff
@@ -2,25 +2,36 @@
...
-#         command => "iptables -A INPUT -s %{clientip} -j DROP"
+#       if [type] == "abuse" {
+#         exec {
+#           command => "iptables -A INPUT -s %{clientip} -j DROP"
+#         }
 #       }
 #     }
 #
-# Run subprocesses via system ruby function
+# WARNING: If you want it non-blocking you should use `&` or dtach or
```
Comment (untergeek) [inline lib/logstash/outputs/exec.rb]: Perhaps make this `dtach` to indicate it is a command-line process.

**Comment by untergeek** [comment]
Thanks for doing this, @magnusbaeck 

Aside from the two comments here (which are subject to debate), it LGTM™

File: lib/logstash/outputs/exec.rb (hunk 1 above)
Comment (magnusbaeck) [inline lib/logstash/outputs/exec.rb]: Good point; backticking all names of commands.

**Pull request merged**